packages/core/src/shared/crashMonitoring.ts

```diff
@@ -515,7 +521,13 @@ export class FileSystemState {
 
     private async deleteHeartbeatFile(ext: ExtInstanceId | ExtInstance, ctx: string) {
         // IMPORTANT: Must use NodeFs here since this is used during shutdown
-        const func = () => nodeFs.rm(this.makeStateFilePath(ext), { force: true })
+        const func = async () => {
+            const filePath = this.makeStateFilePath(ext)
+            // Even when deleting a file, if there is an open file handle it may still exist. This empties the
+            // contents, so that any following reads will get no data.
+            await nodeFs.writeFile(filePath, '')
```

Comment:
use truncate()? https://nodejs.org/api/fs.html#fstruncatepath-len-callback

Reply:
Found we can use `open()` with a `w` flag to do this. This also allows us to use `sync()` in this, which I should have done initially.